adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m

```diff
@@ -108,10 +109,7 @@ - (void)bannerViewDidError:(UADSBannerView *)bannerView error:(UADSBannerError *
   id<GADMAdNetworkConnector> strongConnector = _connector;
   id<GADMAdNetworkAdapter> strongAdapter = _adapter;
   if (strongConnector && strongAdapter) {
-    NSString *errorMsg =
-        [NSString stringWithFormat:@"Unity Ads banner failed to load with error: %@",
-                                   error.localizedDescription];
-    [strongConnector adapter:strongAdapter didFailAd:GADUnityErrorWithDescription(errorMsg)];
+    [strongConnector adapter:strongAdapter didFailAd:error];
```

Comment:
In this case UADSBannerError is already an NSError, and this error is passed from Unity's SDK to Google SDK.

Reply:
@jindoujiao the approach is to send back as granular an error as possible. As @chlandacastro mentioned, since UnityAds SDK already has an NSError for banner failure callbacks, we prefer that. If we have to generate our own error, or there's an adapter specific error, we'll use GADMAdapterUnityErrorWithCodeAndDescription.